### PR TITLE
Add container:run command

### PR DESCRIPTION
**Pull request by dmathieu**
We may want to change the syntax a bit.
Closes heroku/heroku-container-registry/issues/28

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,6 +13,7 @@ $ heroku plugins:install heroku-container-registry
 In a directory with a Dockerfile:
 
 ```
+$ heroku container:run web
```
Comment (tt) [inline README.md]: I would maybe keep the readme focused on the minimal thing to deploy.

File: commands/run.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    flags: [
+      {
+        name: 'verbose',
+        char: 'v',
+        hasValue: false
+      },
+      {
+        name: 'arg',
+        hasValue: true,
+        description: 'set build-time variables'
```
Comment (tt) [inline commands/run.js]: I think we should decouple these commands (and remove this flag). It's not great that all commands need to know about building images.

File: commands/run.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  let dockerfiles = Sanbashi.getDockerfiles(process.cwd(), false)
+  let possibleJobs = Sanbashi.getJobs(tag, dockerfiles)
+
+  let jobs = []
+  if (possibleJobs.standard) {
+    possibleJobs.standard.forEach((pj) => { pj.resource = pj.resource.replace(/standard$/, processType)})
+    jobs = possibleJobs.standard || []
+  }
+  if (!jobs.length) {
+    cli.warn('No images to push')
```
Comment (tt) [inline commands/run.js]: "to run"?

File: commands/run.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+  let processType = context.args.shift()
+  let command = context.args
+  let tag = `heroku-local-${processType}`
+  let dockerfiles = Sanbashi.getDockerfiles(process.cwd(), false)
+  let possibleJobs = Sanbashi.getJobs(tag, dockerfiles)
+
+  let jobs = []
+  if (possibleJobs.standard) {
+    possibleJobs.standard.forEach((pj) => { pj.resource = pj.resource.replace(/standard$/, processType)})
```
Comment (tt) [inline commands/run.js]: Where does "standard" come from?

Comment (dmathieu) [reply commands/run.js]: I have no idea. Looks like it's an array method. I've taken this from the push command.

File: commands/run.js (hunk 2 above)
Comment (dmathieu) [reply commands/run.js]: You mean have a `container:build` command and not run here (nor in push) anymore?

File: commands/run.js (hunk 4 above)
Comment (tt) [reply commands/run.js]: I was curious about `resource.replace(/standard$/, processType)` specifically.

Comment (tt) [reply commands/run.js]: Maybe it's copied from another place there too? Can we remove it?

Comment (dmathieu) [reply commands/run.js]: @heroku/cli any idea about this?

Comment (jdx) [reply commands/run.js]: no idea. I looked at the code and it seems that it never gets set from what I can tell.

Comment (dmathieu) [reply commands/run.js]: `standard` come from [here](https://github.com/heroku/heroku-container-registry/blob/master/lib/sanbashi.js#L33).
The lib detects the process name part of `Dockerfile.web`, and defaults to `standard` if there is nothing.
Maybe this should be renamed to `default` to be clearer.

Comment (tt) [reply commands/run.js]: Wouldn't "web" be better as that's the default process type?

File: commands/run.js (hunk 2 above)
Comment (tt) [reply commands/run.js]: Yes. We probably need to keep it part of the push command for compatibility but I think we should separate it for new commands (and eventually for the push command too).

File: commands/run.js (hunk 4 above)
Comment (dmathieu) [reply commands/run.js]: `web` works for me. I'll change it.

**Review by tt — APPROVED** [review]

**Pull request merged**